### DISCUSSION TITLE
How do I calculate the zoom factor to show a PDF in FullScreen or Maximize ?

### DISCUSSION THREAD

**Discussion by rshetye** [pymupdf/PyMuPDF#1316] [Q&A] [answered]
I am building an auto-slideshow utility for PDFs.

Currently I am using self.GetSize() inside pdf_show() and comparing that with dl.rect() to scale the PDF to fill the entire window / screen ( self.Maximize(True) or self.ShowFullScreen(True) ).

I am able to converge to the right scale within 3 pages, which tells me that my initial measurement is wrong or incomplete. Can you tell me the APIs that will help measure the PDF's dimensions and the maximized/fullscreen dimensions so that I can scale accurately from the first page ? Thanks !

**Comment by JorjMcKie** [comment]
[This](https://pymupdf.readthedocs.io/en/latest/faq.html#how-to-zoom-a-clip-to-a-gui-window) should answer your question.

**Comment by rshetye** [comment]
I've already read the page. It doesn't actually give the details e.g. how do I get the WIDTH and HEIGHT values ?

**Comment by JorjMcKie** [comment, marked as answer]
> how do I get the WIDTH and HEIGHT values?

These are values that your GUI manager tells you. You can e.g. ask wxPython what the physical screen size is, similar tkinter.
The code snippet I mentioned indeed works on the basis what the values of WIDTH and HEIGHT are known.
If you say you don't know them from your GUI manager, then I guess I don't really understand your poblem.
